machine.go

```diff
@@ -496,3 +529,15 @@ func convertToStringSlice(field interface{}) []string {
 	}
 	return result
 }
+
+func convertToStringMap(field interface{}) map[string]string {
+	if field == nil {
+		return nil
+	}
+	fieldMap := field.(map[string]interface{})
```

Comment:
If there's ever a chance for `field` to have a different type, this will panic.
But I suspect you can only call this fields passed through the schema validation/coercion, so should be fine (could use a comment though - anyone looking at it later might wonder why don't we care about panicking).